src/main/java/com/google/api/codegen/ProtoFileView.java

```diff
@@ -14,55 +14,65 @@
  */
 package com.google.api.codegen;
 
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.config.InterfaceConfig;
+import com.google.api.codegen.config.MethodConfig;
+import com.google.api.codegen.config.MethodModel;
+import com.google.api.codegen.config.ProtoTypeRef;
 import com.google.api.tools.framework.model.Field;
-import com.google.api.tools.framework.model.Interface;
 import com.google.api.tools.framework.model.MessageType;
-import com.google.api.tools.framework.model.Method;
-import com.google.api.tools.framework.model.Model;
 import com.google.api.tools.framework.model.ProtoFile;
-import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
-import java.util.HashSet;
+import com.google.api.tools.framework.model.TypeRef;
 import java.util.Set;
+import java.util.TreeSet;
 
 /**
  * A file-based view of model, consisting of a strategy for getting the protocol buffer files from
  * which the model was constructed.
  */
-public class ProtoFileView implements InputElementView<ProtoFile> {
+public class ProtoFileView {
```

Comment:
Why is it necessary to break away from this interface? It's surprising that `ProtoFileView` would not implement `InputElementView<ProtoFile>`. Shouldn't mixins be part of the Model via the mixin config aspect?

Reply:
`reroute_to_grpc_interface` would not be included if I depend solely on the `Model`.

The `InputElementView` interface is legacy from pre-MVVM. It remains because client config is still not migrated, but #1724 will change that.